### ISSUE TITLE
Add support for .NET8

### ISSUE THREAD

**Issue by sabifa** [tomchavakis/nuget-license#200]
I have just upgraded my project to .NET 8 and noticed while running my build pipeline in Azure DevOps that it fails because of `dotnet-project-licenses`.

Output:
```
You can invoke the tool using the following command: dotnet-project-licenses
  Tool 'dotnet-project-licenses' (version '2.7.1') was successfully installed.
  You must install or update .NET to run this application.
  
  App: C:\Users\VssAdministrator\.dotnet\tools\dotnet-project-licenses.exe
  Architecture: x64
  Framework: 'Microsoft.NETCore.App', version '7.0.0' (x64)
  .NET location: C:\hostedtoolcache\windows\dotnet
  
  The following frameworks were found:
    8.0.0 at [C:\hostedtoolcache\windows\dotnet\shared\Microsoft.NETCore.App]
  
  Learn more:
  https://aka.ms/dotnet/app-launch-failed
  
  To install missing framework, download:
  https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=7.0.0&arch=x64&rid=win-x64&os=win10
D:\a\1\s\Festivalmanager\Festivalmanager.csproj(28,9): error MSB3073: The command "dotnet-project-licenses" exited with code -2147450730.
```

**Comment by pedroveiga99** [comment]
Yes, the very same thing happened to me. It would be fantastic if support for .NET8 was added.

**Comment by bsuchorowski** [comment]
Same story on our side. We have just introduced a first dotnet 8.0 app.

**Comment by svenclaesson** [comment]
It would be nice with .net8 support but this does not red to be a blocker. 
Just install both .net8 and .net7 sdk side by side

**Comment by sensslen** [comment]
- @sabifa, @pedroveiga99, @svenclaesson, @bsuchorowski please take the time to review #201 so that we can get that in...

**Comment by sensslen** [comment]
I just released https://www.nuget.org/packages/nuget-license. This is a release of my fork of v3.0.0 of this tool. Unfortunately @tomchavakis has been unresponsive for quite some time and the nuget api key expired which means I was unable to release in the original form.

**Comment by belav** [comment]
@sensslen is there any documentation on the changes in 3.0.0? I can't find any of the following CLI options. And looking at the code it seems like 3.0 is completely different than 2.7.1

```bash
-u
-outfile # possibly just need to pipe the output of the command to a file
--projects-filter
--manual-package-information
--packages-filter
```

Alternatively, is it possible to publish a `2.7.2` that adds a net8 target? The project seems to build fine with adding that target.

**Comment by sensslen** [comment]
> @sensslen is there any documentation on the changes in 3.0.0? I can't find any of the following CLI options. And looking at the code it seems like 3.0 is completely different than 2.7.1
> 
> 
> 
> ```bash
> 
> -u
> 
> -outfile # possibly just need to pipe the output of the command to a file
> 
> --projects-filter
> 
> --manual-package-information
> 
> --packages-filter
> 
> ```
> 
> 
> 
> Alternatively, is it possible to publish a `2.7.2` that adds a net8 target? The project seems to build fine with adding that target.

@belav you are right that version 3 is very different. It addresses some long standing issues. The documentation is available on nuget.org. All available command line options are listed. I'm not able to publish the original nuget, as I don't have the api key. Also I see no value in republishing version 2 under my name.

**Comment by belav** [comment]
> Also I see no value in republishing version 2 under my name.

I tried to figure out how we can convert our command to use 3.0.0. I started with the basics and just output some json by pointing both dotnet-project-licenses and nuget-licenses at our root `Microsoft.Build.Traversal` project. The new version 3.0.0 only gets two project licenses and there should be a lot more.

Being that there is no documentation telling me how 2.7.1 differs from 3.0.0, if there was a 2.7.2 published that worked with net8, I would just use that. That is where the value comes in, at least for consumers of the project. Being that you are taking on the project I understand why you may not want to support the older version.

I ended up building a 2.7.2 myself and committing the nupgk to our repo. If anyone else goes that route, `dotnet tool restore` doesn't seem to respect a `NuGet.config`, and you need to run the command with `--add-source`

I also wanted to say thank you for taking on support of this, I do appreciate it.

**Comment by sensslen** [comment]
> > Also I see no value in republishing version 2 under my name.
> 
> I tried to figure out how we can convert our command to use 3.0.0. I started with the basics and just output some json by pointing both dotnet-project-licenses and nuget-licenses at our root `Microsoft.Build.Traversal` project. The new version 3.0.0 only gets two project licenses and there should be a lot more.
> 
> Being that there is no documentation telling me how 2.7.1 differs from 3.0.0, if there was a 2.7.2 published that worked with net8, I would just use that. That is where the value comes in, at least for consumers of the project. Being that you are taking on the project I understand why you may not want to support the older version.
> 
> I ended up building a 2.7.2 myself and committing the nupgk to our repo. If anyone else goes that route, `dotnet tool restore` doesn't seem to respect a `NuGet.config`, and you need to run the command with `--add-source`
> 
> I also wanted to say thank you for taking on support of this, I do appreciate it.

It looks as if you missed the `-t` flag which also prints transitive dependencies. I'm fine if you want to use the old version of the tool. Keep in mind though that the results may not be accurate due to the fact that version 2 uses a homegrown algorithm to determine the version of a nuget package used, where as v3 uses the nuget internal algorithm. This was actually what made me write version 3.

**Comment by trampster** [comment]
@sensslen your new tool crashes when we try and use it and you new repo has no way to report issues.

**Comment by sensslen** [comment]
@trampster Issues are now turned on on the fork. Can you please report the issue you're seeing?

**Comment by sabifa** [comment]
> I just released https://www.nuget.org/packages/nuget-license. This is a release of my fork of v3.0.0 of this tool. Unfortunately @tomchavakis has been unresponsive for quite some time and the nuget api key expired which means I was unable to release in the original form.

Thank you for your work! Unfortunately `nuget-license` is, as of now, not a suitable replacement for this package. There are core features of this package missing, for example:
* --convert-html-to-text (closed issue as wont do)
* ability to define an output folder/file

**Comment by sensslen** [comment]
@sabifa feel free to submit Pull Requests in order to improve here: https://github.com/sensslen/nuget-license.

**Comment by Suneeh** [comment]
I would love to see a new version of this, since I updated my repos to .NET8 my Pipeline run fails.

**Comment by sensslen** [comment]
> I would love to see a new version of this, since I updated my repos to .NET8 my Pipeline run fails.

@Suneeh There you go: https://www.nuget.org/packages/nuget-license

**Comment by Suneeh** [comment]
@sensslen thanks.. doesn't quite work for me, can you provide an example of an --override-package-information json? Or does it not take a json file?

**Comment by sensslen** [comment]
> @sensslen thanks.. doesn't quite work for me, can you provide an example of an --override-package-information json? Or does it not take a json file?

https://github.com/sensslen/nuget-license/blob/main/.github/workflows/assets/Tests/overwritePackageInformation.json

**Comment by Suneeh** [comment]
Yea just found that as well ! 
Thanks though. I might consider it after rewriting my projects to this nuget package ^^
